v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,197 @@
+const sha256 = require("sha256");
+const { BASE_ENDPOINT, ENDPOINTS } = require("./config");
+const {
+  defaultRequestConfig,
+  CustomError,
+  isDefinedAndNotNullAndNotEmpty,
+  defaultPutRequestConfig,
+  removeUndefinedAndNullValues,
+  getSuccessRespEvents,
+  getErrorRespEvents
+} = require("../../util");
+
+const {
+  getAccessToken,
+  populateIncludes,
+  populateExcludes
+} = require("./util");
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach(element => {
+      const keys = Object.keys(element);
+      keys.forEach(key => {
+        if (key === audienceType) {
+          listType = audienceType;
+        }
+      });
+      if (!listType) {
+        throw new CustomError(
+          `Audience type ${audienceType} not provided`,
+          400
+        );
+      }
+      if (hashRequired) {
+        seedList.push(sha256(element[audienceType]));
```

Comment:
It is risky to use webapp config to extract properties, the `audienceType` should be verified if it belongs to a correct one and constant enums should be used

Reply:
updated